test/resources/vw/io_utils.py

```diff
@@ -0,0 +1,146 @@
+import json
+import logging
+import boto3
+from pathlib import Path
+import shutil
+import os
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+
+def validate_experience(experience):
+    keys = ["observation", "prob", "action", "reward"]
+    for key in keys:
+        is_valid = key in experience
+        if not is_valid:
+            return False
+    return True
+
+
+class CSVReader():
+    """Reader object that loads experiences from CSV file chunks.
+    The input files will be read from in an random order."""
+
+    def __init__(self, input_files):
+        self.files = input_files
+
+    def get_iterator(self):
+        for file in self.files:
+            reader = pd.read_csv(file, chunksize=1000)
+            for df in reader:
+                df_no_nans = df.dropna()
+                for line in df_no_nans.iterrows():
+                    line_dict = line[1].to_dict()
+                    yield line_dict
+
+
+class JsonLinesReader():
```

Comment:
Is this class still needed? We moved from json to csv

Reply:
that’s not needed right now, but we can leave it there for future use cases.